### ISSUE TITLE
Sitemap Label Colour not working

### ISSUE THREAD

**Issue by Dash** [openhab/openhab-webui#2082]
## Which UI are you reporting an issue for?

- [X] Basic UI
- [ ] HABPanel
- [ ] HABot
- [ ] CometVisu

## The problem

Conditional Label colours no longer work on Sitemaps, though simply specifying a colour does work.  I'm not sure at which point this regressed.

This setup
![image](https://github.com/openhab/openhab-webui/assets/129838/62e690f4-cba0-4987-99c5-964489aa6abf)

Produces:
![image](https://github.com/openhab/openhab-webui/assets/129838/25a95ea5-921c-4437-a100-176fb7a92c0b)

The UI generates this code:

```javascript
sitemap page_c6c037636f label="New Sitemap" {
    Text item=KitchenLights labelcolor=[KitchenLights == ON =" red"] label="Lights are: [%s]"
}
```

I have checked in Firefox, Chrome, and through the Android App.

## Expected behavior

I'm expecting the above example to show the text "Kitchen Lights" in red on sitemap.

## Steps to reproduce

1. Build a new site map
2. Add a label to a switch
3. Add a conditional colour based on the switch status

## Your environment

```yaml
runtimeInfo:
  version: 4.1.0.M1
  buildString: Milestone Build
locale: en-GB
systemInfo:
  configFolder: /etc/openhab
  userdataFolder: /var/lib/openhab
  logFolder: /var/log/openhab
  javaVersion: 17.0.8
  javaVendor: N/A
  osName: Linux
  osVersion: 5.14.21-150400.24.81-default
  osArchitecture: amd64
  availableProcessors: 2
  freeMemory: 332692888
  totalMemory: 778043392
  uptime: 353815
  startLevel: 100
addons:
  - automation-jsscripting
  - binding-astro
  - binding-exec
  - binding-gpstracker
  - binding-http
  - binding-icalendar
  - binding-mqtt
  - binding-network
  - binding-snmp
  - binding-systeminfo
  - binding-xmppclient
  - persistence-rrd4j
  - transformation-jsonpath
  - transformation-map
  - transformation-regex
  - ui-basic
  - ui-habpanel
clientInfo:
  device:
    ios: false
    android: false
    androidChrome: false
    desktop: true
    iphone: false
    ipod: false
    ipad: false
    edge: false
    ie: false
    firefox: true
    macos: false
    windows: true
    cordova: false
    phonegap: false
    electron: false
    nwjs: false
    webView: false
    webview: false
    standalone: false
    os: windows
    pixelRatio: 1
    prefersColorScheme: dark
  isSecureContext: true
  locationbarVisible: true
  menubarVisible: true
  navigator:
    cookieEnabled: true
    deviceMemory: N/A
    hardwareConcurrency: 12
    language: en-GB
    languages:
      - en-GB
      - en
    onLine: true
    platform: Win32
  screen:
    width: 3840
    height: 2160
    colorDepth: 24
  support:
    touch: false
    pointerEvents: true
    observer: true
    passiveListener: true
    gestures: false
    intersectionObserver: true
  themeOptions:
    dark: dark
    filled: true
    pageTransitionAnimation: default
    bars: light
    homeNavbar: default
    homeBackground: default
    expandableCardAnimation: default
  userAgent: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101
    Firefox/117.0
timestamp: 2023-09-21T17:37:20.326Z

```

## Browser console

```txt

```

## Browser network traffic

![image](https://github.com/openhab/openhab-webui/assets/129838/f78d5485-3f99-48c4-867e-fabfd7986c78)

**Comment by lolodomo** [comment]
In your exemple, I ser an  unexpected space before the color. Try replacing " red" by "red".

**Comment by Dash** [comment]
> In your exemple, I ser an unexpected space before the color. Try replacing " red" by "red".

You'll note that this is done through the sitemap builder, where there are no quotes.  But I tried stripping out just the space between the = and red, and that didn't make a difference, but taking it a step further, I stripped out all the whitespace:
![image](https://github.com/openhab/openhab-webui/assets/129838/30aa28cc-8ea1-4959-8e55-ccfdc1a1f603)

And that has worked:
![image](https://github.com/openhab/openhab-webui/assets/129838/5c7b7764-6e88-40cd-8341-de232ec38c0c)

I'd classify that this is a workaround, as this wasn't the behaviour before.

**Comment by lolodomo** [comment]
Maybe a regression with recently merged PR #2045 ? I see something changed about spaces in the description. 
@mherwege

**Comment by mherwege** [comment]
> Maybe a regression with recently merged PR #2045 ? I see something changed about spaces in the description.
> @mherwege

Unlikely, as this only touched the parser to convert from the sitemap code to the internal representation. The space appears when converting from the UI presentation to the internal and text representation. I will have a look at it.

**Comment by mherwege** [comment]
It is probably a regression from https://github.com/openhab/openhab-core/pull/3769, in itself created to solve https://github.com/openhab/openhab-webui/pull/1487.
